src/riot-rs-embassy/src/arch/nrf52.rs

```diff
@@ -21,18 +10,30 @@ unsafe fn SWI0_EGU0() {
 #[cfg(feature = "usb")]
 pub mod usb {
     use embassy_nrf::{
-        peripherals,
-        usb::{vbus_detect::HardwareVbusDetect, Driver},
+        bind_interrupts, peripherals, rng,
+        usb::{
+            self,
+            vbus_detect::{self, HardwareVbusDetect},
+            Driver,
+        },
     };
 
     use crate::arch;
 
+    bind_interrupts!(struct Irqs {
+        USBD => usb::InterruptHandler<peripherals::USBD>;
+        POWER_CLOCK => vbus_detect::InterruptHandler;
+        RNG => rng::InterruptHandler<peripherals::RNG>;
+    });
+
     pub type UsbDriver = Driver<'static, peripherals::USBD, HardwareVbusDetect>;
 
     pub fn driver(peripherals: &mut arch::OptionalPeripherals) -> UsbDriver {
-        use super::Irqs;
-
         let usbd = peripherals.USBD.take().unwrap();
         Driver::new(usbd, Irqs, HardwareVbusDetect::new(Irqs))
     }
 }
+
+pub fn init(config: Config) -> OptionalPeripherals {
+    OptionalPeripherals::from(embassy_nrf::init(config))
```

Comment:
```suggestion
    let peripherals = embassy_nrf::init(config));
    OptionalPeripherals::from(peripherals)
```

IMO otherwise, the much more importand `init()` call is hidden inside the textually opulent `OptionalPeripherals::from(...)`.